lib/models/index.ts

```diff
@@ -32,6 +33,19 @@ export class Models extends Construct {
     super(scope, id);
 
     const models: SageMakerModelEndpoint[] = [];
+    let schedulerRole: iam.Role | undefined = undefined;
+
+    if (
+      props.config.llms?.sagemakerSchedule?.enabled
+    ) {
+        schedulerRole = new iam.Role(this, 'SchedulerRole', {
+        assumedBy: new iam.ServicePrincipal('scheduler.amazonaws.com'),
+        description: 'Role for Scheduler to interact with SageMaker',
+      });
+      
+      schedulerRole.addManagedPolicy(iam.ManagedPolicy.fromAwsManagedPolicyName('AmazonSageMakerFullAccess'));
+      this.suppressCdkNagWarningForEndpointRole(schedulerRole);
```

Comment:
`IAM4` and `IAM5` are typically pretty broad ones to disable without a specific `appliesTo` clause (if someone else grabs the role elsewhere in a stack and adds or changes it, the rule is already modified). I would consider being more careful with that suppression.